### DISCUSSION TITLE
[Question] Multiple customers in same project

### DISCUSSION THREAD

**Discussion by mauroalexandre** [frangoteam/FUXA#469] [General]
Is possible I'm use the same server installation for multiple customers, using cookies or something?

I'm found **FUXA/client/src/app/_services/rcgi/resource-storage** and **AppID**, this is the way? how?

Thanks alot.

**Comment by unocelli** [comment]
Hi,
No, `rcgi` is to separate the diverse communication interface from frontend to backend :
- `ResDemoService` used in demo version [https://frangoteam.github.io/](https://frangoteam.github.io/) that don't have the FUXA backend
- `ResClientService` used bei project without FUXA backend. The communication is done via a bridge and the FUXA frontend should be build as web component
- `ResWebApiService` is the standard with WebAPI

You should use docker or the user authorization.

**Comment by mauroalexandre** [comment]
So, I need one instance for each customer?

For example, I have 3 customers, I need to run 3 diferent docker instances and put in different URI, and use HTTP Access for each URI.

**Comment by unocelli** [comment]
You could use the github installation in different folder and change only the web server port...should work.
I will move the issue to discussions

**Comment by MatthewReed303** [comment]
With Ignition you can run as many projects as you like and assign new user group with all the sub group permissions for that project. To access that project it just the URL name of the project, all running on the same port and backend. 

Something like this would be very helpful and useful for IIOT applications etc even the ability to create your own groups and permissions etc 

So basically at the login screen, the user that belongs to the project group automatically gets redirected to the project URL in background

**Reply by mauroalexandre** [reply]
Hi @MatthewReed303 , what is Ignition?

**Reply by MatthewReed303** [reply]
Hi @mauroalexandre see here: https://inductiveautomation.com/scada-software/ 

It's a SCADA platform called Ignition and the web part is called perspective view. It's a great platform for very large projects but just small projects like running a pump etc it's very very expensive.

**Reply by mauroalexandre** [reply]
Let's make the fuxa have the same functions!

I'm ready!

**Reply by MatthewReed303** [reply]
Fuxa has an awesome base to build on and add the functionality. Where Ignition shines is data handling with large amounts of data and data bases. Having a connector in Fuxa to connect to all types of DBs and use SQL commands via the scripting interface will improve the functionality a lot. If the existing table interface could connect to DB and read data directly from the DB and allow optional input. ( Indusoft SCADA was great for this ) 

With reports being able to edit the PDF and a way to embed Tags directly into the report with your own custom layout and agility to trigger printing etc by a tag.

Interface to create/import custom SVG symbols with animations etc You can select every element of the SVG and edit it's effects/animations 

A good interface for handling recipes via DB would be a great addition. 

The list goes on, but I guess a discussion should be created and a list of all major features people would like with voting

**Reply by unocelli** [reply]
Hi, interesting...i think it could be an option, but don't forget that the concept of fuxa is to have everything in one application: development environment and the runtime application. and the runtime uses only one project.

**Reply by MatthewReed303** [reply]
Hi @unocelli could you not use the same editor but have an option to select which project you want to edit? Each project runs independently with it's own users defined by URLs example fuxa/project1/home, fuxa/project2/home, fuxa/project3/home each project can be customized differently for each customer. Basically at the login page will just be a redirect, User1 belongs to project1 so the automatically get directed to there project.

**Reply by unocelli** [reply]
@MatthewReed303 I cannot understand the gain. currently this is already possible, you have to configure the authorisations, on the access points of your views
